lib/python/picongpu/utils/memory_calculator.py

```diff
@@ -0,0 +1,368 @@
+            target_n_y=None,
+            target_n_z=None,
+            num_additional_attributes=0,
+            particles_per_cell=2,
+            sim_dim=3
+    ):
+        """
+        Memory reserved for all particles of a species on a GPU.
+        We currently neglect the constant species memory.
+
+        Parameters
+        ----------
+
+        target_n_x : int
+            number of cells in x direction containing the target
+        target_n_y : int
+            number of cells in y direction containing the target
+        target_n_z : int
+            number of cells in z direction containing the target
+        num_additional_attributes : int
+            number of additional attributes like e.g. ``boundElectrons``
+        particles_per_cell : int
+            number of particles of the species per cell
+        sim_dim : int
+            simulation dimension (available for PIConGPU: 2 and 3)
+
+        Returns
+        -------
+
+        req_mem : int
+            required memory {unit: bytes} per GPU and species
+        """
+
+        if target_n_x is None:
+            target_n_x = self.n_x
+        if target_n_y is None:
+            target_n_y = self.n_y
+        if target_n_z is None:
+            target_n_z = self.n_z
+
+        # memory required by the standard particle attributes
+        standard_attribute_mem = np.array([
+            3 * self.value_size,  # momentum
+            sim_dim * self.value_size,  # position
+            1,  # multimask (``uint8_t``)
+            2,  # cell index in supercell (``typedef uint16_t lcellId_t``)
+            1 * self.value_size  # weighting
+        ])
+
+        # memory per particle for additional attributes {unit: byte}
+        additional_mem = num_additional_attributes * self.value_size
+        # \TODO we assume value_size here - that could be different
+        # cells filled by the target species
+        local_cells = target_n_x * target_n_y * target_n_z
+
+        req_mem = local_cells * (np.sum(
+            standard_attribute_mem) + additional_mem) * particles_per_cell
+        return req_mem
+
+    def mem_req_by_rng(
+            self,
+            n_x=None,
+            n_y=None,
+            n_z=None,
+    ):
+        """
+        Memory reserved for the random number generator state on each GPU.
+        The RNG we use is: MRG32k3a
+
+        Parameters
+        ----------
+        n_x : int
+            number of cells in x direction (per accelerator)
+        n_y : int
+            number of cells in y direction (per accelerator)
+        n_z : int
+            number of cells in z direction (per accelerator)
+
+        Returns
+        -------
+
+        req_mem : int
+            required memory {unit: bytes} per accelerator
+        """
+        if n_x is None:
+            n_x = self.n_x
+        if n_y is None:
+            n_y = self.n_y
+        if n_z is None:
+            n_z = self.n_z
+
+        req_mem_per_cell = 6 * 8  # bytes
```

Comment:
The default RNG on GPU is `XorMin` and use `6 * 4 byte` per cell on CORE + BORDER (GUARD has currently no rng state)

Reply:
`6 * 8 byte` is only correct for `MRG32k3aMin` (not the default)

Reply:
Perfect, thanks!

**offline** with @psychocoderHPC:
the default for CPU and `AlpakaRand` is TinyMT with a state size of `7 * 4` bytes